src/main/java/com/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java

```diff
@@ -0,0 +1,113 @@
+package com.internet.shop.dao.jdbc;
+
+import com.internet.shop.dao.ProductDao;
+import com.internet.shop.exceptions.DataProcessingException;
+import com.internet.shop.lib.Dao;
+import com.internet.shop.model.Product;
+import com.internet.shop.util.ConnectionUtil;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class ProductDaoJdbcImpl implements ProductDao {
+
+    @Override
+    public Product create(Product product) {
+        String query = "INSERT INTO products (name, price) VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection
+                        .prepareStatement(query, Statement.RETURN_GENERATED_KEYS)) {
+            setValues(preparedStatement, product);
+            preparedStatement.executeUpdate();
+            ResultSet generatedKeys = preparedStatement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                product.setId(generatedKeys.getLong(1));
+            }
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Can't insert product", ex);
+        }
+        return product;
+    }
+
+    @Override
+    public Product update(Product product) {
+        String query = "UPDATE products SET name =?,price=? WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement preparedStatement = connection
+                        .prepareStatement(query)) {
+            setValues(preparedStatement, product);
+            preparedStatement.setLong(3, product.getId());
+            if (preparedStatement.executeUpdate() > 0) {
+                return product;
+            }
+            throw new DataProcessingException("Can't update product");
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Can't update product", ex);
+        }
+    }
+
+    @Override
+    public Optional<Product> get(Long id) {
+        String query = "SELECT * FROM products WHERE id = ? AND is_deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection
+                        .prepareStatement(query)) {
+            preparedStatement.setLong(1, id);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            if (resultSet.next()) {
+                return Optional.of(conversionToProduct(resultSet));
+            }
+            throw new DataProcessingException("Can't get product with id: " + id);
```

Comment:
since we return optional here just return Optional.empty() instead of throwing exception (it will be thrown on the service layer)

Reply:
done